{
    "body": "### Steps to reproduce\n\n(Fyi, I am a new user of `spack`, so I may not know what I am doing.)\r\n\r\nIn a fresh Alpine Linux 3.14.2 chroot environment (and probably also in a native install of Alpine Linux), I run the following as root:\r\n\r\n    #  apk  update\r\n    #  apk  add  build-base  bash  git  python3\r\n\r\nI then run the following as a non-root user:\r\n\r\n    $  git  clone  -c feature.manyFiles=true  https://github.com/spack/spack.git\r\n    $  .  spack/share/spack/setup-env.sh\r\n    $  time  spack  --debug  --stacktrace  spec  zlib\r\n\r\n(See error message below.)\r\n\r\nIf I install Alpine's `py3-clingo` package, then `spack` appears to run successfully, but `spack` prints warnings about the version of `clingo` it is using.\r\n\r\n(See warning below.)\r\n\r\nI can find similar bug reports for other OSes.  However, (1) they are marked as closed, (2) the issue is still occurring with Alpine Linux.\n\n### Error message\n\n```\r\n$  time  spack  --debug  --stacktrace  spec  zlib\r\nlib/spack/spack/cmd/__init__.py:122 ==> [2021-09-30-22:01:59.390481] Imported spec from built-in commands\r\nlib/spack/spack/cmd/__init__.py:122 ==> [2021-09-30-22:01:59.391952] Imported spec from built-in commands\r\nInput spec\r\n--------------------------------\r\nzlib\r\n\r\nConcretized\r\n--------------------------------\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.396546] Reading config file /home/parke/spack/etc/spack/defaults/config.yaml\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.442425] Reading config file /home/parke/spack/etc/spack/defaults/bootstrap.yaml\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.456440] Reading config file /home/parke/spack/etc/spack/defaults/config.yaml\r\nlib/spack/spack/database.py:373 ==> [2021-09-30-22:01:59.484461] DATABASE LOCK TIMEOUT: 3s\r\nlib/spack/spack/database.py:377 ==> [2021-09-30-22:01:59.484919] PACKAGE LOCK TIMEOUT: No timeout\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.485813] Reading config file /home/parke/spack/etc/spack/defaults/bootstrap.yaml\r\nlib/spack/spack/database.py:373 ==> [2021-09-30-22:01:59.497010] DATABASE LOCK TIMEOUT: 3s\r\nlib/spack/spack/database.py:377 ==> [2021-09-30-22:01:59.497480] PACKAGE LOCK TIMEOUT: No timeout\r\nlib/spack/spack/bootstrap.py:476 ==> [2021-09-30-22:01:59.497931] [BOOTSTRAP CONFIG SCOPE] name=_builtin\r\nlib/spack/spack/bootstrap.py:488 ==> [2021-09-30-22:01:59.498692] [BOOTSTRAP CONFIG SCOPE] name=defaults, path=/home/parke/spack/etc/spack/defaults\r\nlib/spack/spack/bootstrap.py:489 ==> [2021-09-30-22:01:59.499041] [BOOTSTRAP CONFIG SCOPE] name=defaults/linux, path=/home/parke/spack/etc/spack/defaults/linux\r\nlib/spack/spack/bootstrap.py:488 ==> [2021-09-30-22:01:59.499366] [BOOTSTRAP CONFIG SCOPE] name=system, path=/etc/spack\r\nlib/spack/spack/bootstrap.py:489 ==> [2021-09-30-22:01:59.499721] [BOOTSTRAP CONFIG SCOPE] name=system/linux, path=/etc/spack/linux\r\nlib/spack/spack/bootstrap.py:488 ==> [2021-09-30-22:01:59.500058] [BOOTSTRAP CONFIG SCOPE] name=site, path=/home/parke/spack/etc/spack\r\nlib/spack/spack/bootstrap.py:489 ==> [2021-09-30-22:01:59.500307] [BOOTSTRAP CONFIG SCOPE] name=site/linux, path=/home/parke/spack/etc/spack/linux\r\nlib/spack/spack/bootstrap.py:488 ==> [2021-09-30-22:01:59.500582] [BOOTSTRAP CONFIG SCOPE] name=user, path=/home/parke/.spack\r\nlib/spack/spack/bootstrap.py:489 ==> [2021-09-30-22:01:59.500886] [BOOTSTRAP CONFIG SCOPE] name=user/linux, path=/home/parke/.spack/linux\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.502088] Reading config file /home/parke/.spack/linux/compilers.yaml\r\nlib/spack/spack/bootstrap.py:556 ==> [2021-09-30-22:01:59.519363] [BOOTSTRAP ROOT SPEC] clingo: clingo-bootstrap@spack+python %gcc target=x86_64\r\nlib/spack/spack/bootstrap.py:374 ==> [2021-09-30-22:01:59.519783] [BOOTSTRAP MODULE clingo] Try importing from Python\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.520439] Reading config file /home/parke/spack/etc/spack/defaults/bootstrap.yaml\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:01:59.687723] Reading config file /home/parke/spack/etc/spack/defaults/config.yaml\r\nlib/spack/spack/bootstrap.py:94 ==> [2021-09-30-22:02:33.614824] [BOOTSTRAP MODULE clingo] The installed spec \"clingo-bootstrap@spack+python %gcc target=x86_64 ^python@3.9 /hmnv6gk5wha64k6r3s7hid35mzvhkuot\" provides the \"clingo\" Python module\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:02:33.615723] Reading config file /home/parke/spack/etc/spack/defaults/repos.yaml\r\nlib/spack/spack/config.py:991 ==> [2021-09-30-22:02:33.620863] Reading config file /home/parke/spack/etc/spack/defaults/config.yaml\r\nTraceback (most recent call last):\r\n  File \"/home/parke/spack/bin/spack\", line 100, in <module>\r\n    sys.exit(spack.main.main())\r\n  File \"/home/parke/spack/lib/spack/spack/main.py\", line 797, in main\r\n    return _invoke_command(command, parser, args, unknown)\r\n  File \"/home/parke/spack/lib/spack/spack/main.py\", line 525, in _invoke_command\r\n    return_val = command(parser, args)\r\n  File \"/home/parke/spack/lib/spack/spack/cmd/spec.py\", line 109, in spec\r\n    spec.concretize()\r\n  File \"/home/parke/spack/lib/spack/spack/spec.py\", line 2624, in concretize\r\n    self._new_concretize(tests)\r\n  File \"/home/parke/spack/lib/spack/spack/spec.py\", line 2594, in _new_concretize\r\n    result = spack.solver.asp.solve([self], tests=tests)\r\n  File \"/home/parke/spack/lib/spack/spack/solver/asp.py\", line 1691, in solve\r\n    return driver.solve(setup, specs, dump, models, timers, stats, tests)\r\n  File \"/home/parke/spack/lib/spack/spack/solver/asp.py\", line 332, in solve\r\n    self.control = clingo.Control()\r\nAttributeError: module 'clingo' has no attribute 'Control'\r\nCommand exited with non-zero status 1\r\nreal    0m 38.79s\r\nuser    0m 35.99s\r\nsys     0m 2.07s\r\n```\r\n\r\nAnd here is the warning I see if I install Alpine's `py3-clingo` package:\r\n\r\n```\r\n==> Warning: Spec clingo-bootstrap@spack%gcc@9.3.0~docs~ipo+python build_type=Release arch=linux-rhel5-x86_64 ^python@3.9%gcc@9.3.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87 arch=linux-rhel5-x86_64 did not provide module clingo\r\n```\n\n### Information on your system\n\n$  spack  debug  report\r\n* **Spack:** 0.16.3-4547-7694c58736\r\n* **Python:** 3.9.5\r\n* **Platform:** linux-alpine3-haswell\r\n* **Concretizer:** clingo\r\n\r\nFyi, I am running the Alpine Linux chroot inside of an Lxroot virtual environment, but I don't think that makes a difference.  I suspect the same problem will be encountered on any system running Alpine Linux.\n\n### General information\n\n- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform\n- [X] I have searched the issues of this repo and believe this is not a duplicate\n- [X] I have run the failing commands in debug mode and reported the output",
    "user": "parke",
    "url": "https://api.github.com/repos/spack/spack/issues/26394",
    "updated_at": "2021-10-02 11:23:44",
    "created_at": "2021-09-30 22:14:07",
    "closed_at": "None",
    "state": "open",
    "title": "Error: module 'clingo' has no attribute 'Control' when bootstrapping Spack on Alpine Linux",
    "number": 26394,
    "milestone": null,
    "labels": [
        "bug",
        "triage",
        "bootstrap"
    ],
    "id": 1012671156,
    "html_url": "https://github.com/spack/spack/issues/26394",
    "assignees": [
        "alalazo"
    ],
    "comments": 0
}